frontend/src/pages/DataCatalog/dataset_explorer/DatasetExplorer.tsx

```diff
@@ -43,33 +46,17 @@ function DatasetExplorer(props: { preFilterDatasetIds: string[] }) {
     [NAME_FILTER_ID]: props.preFilterDatasetIds,
   });
 
-  // Once the metadata is loaded, update the filter to only include valid
-  // dataset ids
-  useOnMetadataLoaded((metadata) => {
-    const validIds = props.preFilterDatasetIds.filter(
-      (datasetId) => !!metadata[datasetId]
-    );
-    setActiveFilter((prevFilter) => {
-      const names = prevFilter[NAME_FILTER_ID];
-      const newNames = names.filter((name) => validIds.includes(name));
-      return {
-        ...prevFilter,
-        [NAME_FILTER_ID]: newNames,
-      };
-    });
-  });
```

Comment:
What's the reason for getting rid of this?

Reply:
Sorry I actually changed preFilterDatasetIds to be source IDs but forgot to rename. I didn't think we needed to check against the dataset metadata anymore since all the data sources are in static config.

Reply:
Oh okay! I think you can delete useOnMetadataLoaded in that case, since I'm pretty sure it's only used here